x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/sample_logs_input.test.tsx

```diff
@@ -27,6 +32,145 @@ const changeFile = async (input: HTMLElement, file: File) => {
+    expect(() => parseNDJSON(validNDJSONWithSpaces, true)).toThrow();
+  });
+
+  it('should not parse multiline NDJSON in single-line mode', () => {
+    expect(() => parseNDJSON(multilineNDJSON, false)).toThrow();
+  });
+
+  it('should parse multiline NDJSON in multiline mode', () => {
+    expect(parseNDJSON(multilineNDJSON, true)).toEqual(content);
+  });
+
+  it('should parse single-line JSON Array', () => {
+    expect(parseNDJSON(singlelineArray, false)).toEqual([content]);
+    expect(parseNDJSON(singlelineArray, true)).toEqual([content]);
+  });
+
+  it('should not parse a multi-line JSON Array', () => {
+    expect(() => parseNDJSON(multilineArray, false)).toThrow();
+    expect(() => parseNDJSON(multilineArray, true)).toThrow();
+  });
+
+  it('should parse single-line JSON with one entry', () => {
+    const fileContent = '{"message":"test message 1"}';
+    expect(parseNDJSON(fileContent)).toEqual([{ message: 'test message 1' }]);
+  });
+
+  it('should handle empty content', () => {
+    expect(parseNDJSON('  ', false)).toEqual([]);
+    expect(parseNDJSON('  ', true)).toEqual([]);
+  });
+
+  it('should handle empty lines in file content', () => {
+    const fileContent = '\n\n{"message":"test message 1"}\n\n{"message":"test message 2"}\n\n';
+    expect(parseNDJSON(fileContent, false)).toEqual(content);
+    expect(parseNDJSON(fileContent, true)).toEqual(content);
+  });
+});
+
+describe('parseJSONArray', () => {
+  const content = [{ message: 'test message 1' }, { message: 'test message 2' }];
+  const singlelineArray = '[{"message":"test message 1"},{"message":"test message 2"}]';
+  const multilineArray = '[{"message":"test message 1"},\n{"message":"test message 2"}]';
+  const multilineWithSpacesArray =
+    '   [ \n\n{"message":  "test message 1"},\n{"message"   :\n\n"test message 2"}\n]\n';
+  const malformedJSON = '[{"message":"test message 1"}';
+
+  it('should parse valid JSON array', () => {
+    const expected = {
+      entries: content,
+      pathToEntries: '',
+      errorNoArrayFound: false,
+    };
+    expect(parseJSONArray(singlelineArray)).toEqual(expected);
+    expect(parseJSONArray(multilineArray)).toEqual(expected);
+    expect(parseJSONArray(multilineWithSpacesArray)).toEqual(expected);
+  });
+
+  it('should parse valid JSON object with array entries', () => {
+    const expected = {
+      entries: content,
+      pathToEntries: 'events',
+      errorNoArrayFound: false,
+    };
+    expect(parseJSONArray(complexEventsJSON)).toEqual(expected);
+  });
+
+  it('should fail if the JSON object with array entries has not an identifier-like key', () => {
+    const expected = {
+      entries: [],
+      pathToEntries: '',
+      errorNoArrayFound: true,
+    };
+    expect(parseJSONArray(nonIdentifierLikeKeyInJSON)).toEqual(expected);
+  });
+
+  it('should return error for JSON that does not contain an array', () => {
+    const fileContent = '{"records" : {"message": "test message 1"}}';
```

Comment:
Why is this an error? Isn't it same scenario as NDJSON with one record , but inside a key?

Reply:
Since we considered one line NDJSON as a valid scenario , this should as well be.

Reply:
Oh wait, all's working correctly. This is a test for `parseJSONArray` function, it is correct that it does not parse NDJSON. I'll add this as a passing test for the main function as well.